Sources/BraveWallet/Crypto/Stores/CryptoStore.swift

```diff
@@ -66,6 +66,18 @@ public class CryptoStore: ObservableObject {
   @Published private(set) var pendingRequest: PendingRequest? {
     didSet {
       if pendingRequest == nil {
+        /*
+         We need to check if Tx Confirmation modal is ready
+         to be dismissed. It could be not ready as there is no pending tx
+         but an active tx is being shown its different state like
+         loading, submitted, completed or failed.
+         As such we need to continue displaying Tx Confirmation until the
+         user taps Ok/Close on the status overlay.
+         */
+        if let confirmationStore = self.confirmationStore, !confirmationStore.isReadyToBeDismissed {
+          pendingRequest = .transactions([])
+          return
+        }
```

Comment:
Should this logic just exist in `prepare()`? We're checking if we're ready to dismiss in `prepare()` and if we're not ready we're assigning `pendingRequest` to nil, then in this didSet we're checking the same conditions again and assigning `pendingRequest` to `.transaction([])`?